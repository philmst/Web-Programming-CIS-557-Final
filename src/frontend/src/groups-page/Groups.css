html,
body {
  margin: 0;
  padding: 0;
}

/* header styles*/

.header {
  background: rgb(0,31,84);
  height: 12vh;
  width: 100vw;
  
  display: flex;
  flex-direction: row;
  justify-content: left;
  align-items: center;
} 

.social-media-title {
  border: 0.5px solid rgb(10, 17, 40);
  height: 100%;
  width: 75vw;
  
  font-size: 56px;
  padding: 0px;
  color: #fff;
} 

.profile-picture {
  background: rgb(3,64,120);
  height: 100%;
  width: 10vw;
  
  display: flex;
  justify-content: center;
  align-items:center;
} .img {
  border: 0.5px solid rgb(10, 17, 40);
  background-color: #fff;
  
  height: 60px;
  width: 60px;
  border-radius: 60px;
} 

.username {
  border-left: 0.5px solid rgb(10, 17, 40);
  height: 100%;
  width: 15vw;
  
  font-size: 28px;
  padding: 0px;
  color: #fff;
  
  display: flex;
  justify-content: center;
  align-items: center;
}

/* top navbar */

.top-navbar {
  background: rgb(3,64,120);
  border: 0.5px solid rgb(10, 17, 40);
  height: 6vh;
  width: 100vw;
  color: #fff;
  display: flex;
  justify-content: space-between;
}

.home-link {
  border-right: 0.5px solid rgb(10, 17, 40);
  height: 100%;
  width: 15vw;
  
  display: flex;
  justify-content:center;
  align-items:center;
}

.profile-link {
  border-left: 0.5px solid rgb(10, 17, 40);
  height: 100%;
  width: 15vw;
  
  display: flex;
  justify-content:center;
  align-items:center;
}

/* main container */

.main-container {
  display: flex;
  flex-direction: row;
  height: 82vh;
  width: 100vw;
  margin: 0;
  padding: 0;
}

/*side navbar*/

.side-navbar {
  height: 100%;
  width: 15vw;
  background: #fff;
} button {
  border: 0.5px solid rgb(10, 17, 40);
  height: 5vh;
  width: 15vw;
  background: rgb(18,130,162);
  
  display: flex;
  justify-content:center;
  align-items:center;
}

/*updates area*/

.main-area {
  border: 0.5px solid rgb(10, 17, 40);
  height: 100%;
  width: 70vw;
}

.post-area {
  height: 40%;
  width: 70vw;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  gap: 50px;
}

.groups-area {
  border-top: 0.5px solid rgb(10, 17, 40);
  height: 60%;
  width: 70vw;

  gap: 20px 20px;
  overflow-x: hidden;
  overflow-y: scroll;
}
/*live updates*/

.live-updates {
  border-top: 0.5px solid rgb(10, 17, 40);
  background: #fff;
  height: 100%;
  width: 15vw;
}

/* groups style */

.group-container {
  display: flex;
  background-color: #F3EFF5;
  border: 2px solid #454955;
  border-radius: 6px;
  padding: 15px;

  gap: 20px;
  box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.25);
  font-family: "LibertinusSans";
  transition: all 0.2s ease;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 100px;
}

.group-container:hover {
  box-shadow: none;
  filter: contrast(1.15);
}

.group-container:active {
  box-shadow: inset 0px 5px 10px rgba(0, 0, 0, 0.25);
}

.join-button {
  outline: 2px solid rgb(3,64,120);
  border-radius: 4px;
  height: 60px;
  width: 120px;
} 

.join-button:hover {
  box-shadow: none;
  filter: contrast(1.15);
}

.group-info {
  border: 0px solid #454955;
  border-radius: 4px;
 
} ul, li {
  font-size: 18px;
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.non-member-group {
  background-color: white;
  border-color: white;
}