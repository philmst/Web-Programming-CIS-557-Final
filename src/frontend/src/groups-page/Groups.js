import { useEffect, useState, React } from 'react';
import './Groups.css';

const lib = require('./GroupsModule');

function Groups(props) {
  const { changeState } = props;
  // eslint-disable-next-line no-unused-vars
  const [allGroups, setAllGroups] = useState([]);

  const updateGroups = async () => {
    const groups = await lib.getGroups(changeState);

    // eslint-disable-next-line
    console.log(groups.result);
    setAllGroups(groups.result);
    lib.parseGroups(groups.result[0]);
  };

  const createGroup = async () => {
    // eslint-disable-next-line
    const res = await lib.createGroup(changeState,
      document.getElementById('groupname').value,
      1,
      document.getElementById('groupdescription').value,
      document.getElementById('ispublic').value,
      document.getElementById('topic1').value,
      document.getElementById('topic2').value,
      document.getElementById('topic3').value);

    const groups = await lib.getGroups(changeState);

    setAllGroups(groups.result);
    lib.parseGroups(groups.result[0]);
  };

  useEffect(() => { updateGroups(); }, []);

  return (
    <div className="container">
      <div className="header">
        <div className="social-media-title">Social Media</div>
        <div className="profile-picture">
          <div className="img" />
        </div>
        <div className="username">Hi, username!</div>
      </div>

      <div className="top-navbar">
        <div className="home-link">Home</div>
        <div className="profile-link">Profile</div>
      </div>

      <div className="main-container">

        <div className="side-navbar">
          <button type="submit" className="notifications" onClick={() => changeState({ link: '/notifications' })}>Notifications</button>
          <button type="submit" className="events">Events</button>
          <button type="submit" className="groups" onClick={() => changeState({ link: '/groups' })}>Groups</button>
          <button type="submit" className="videos">Videos</button>
          <button type="submit" className="photos">Photos</button>
        </div>

        <div className="main-area">
          <div className="post-area">
            <div className="left-group-input">
              <div className="textDiv" id="groupnameDiv">
                <label htmlFor="groupname">
                  Group Name:
                  <input className="text" id="groupname" type="text" placeholder="group name" />
                </label>
              </div>
              <div className="textDiv" id="groupdescriptionDiv">
                <label htmlFor="groupdescription">
                  Group Description:
                  <input className="text" id="groupdescription" type="text" placeholder="group description" />
                </label>
              </div>
              <div className="textDiv" id="ispublicDiv">
                <label htmlFor="ispublic">
                  Is Public?
                  <input className="text" id="ispublic" type="text" placeholder="1 or 0" />
                </label>
              </div>
            </div>
            <div className="right-group-input">
              <div className="textDiv" id="topic1Div">
                <label htmlFor="topic1">
                  Topic 1:
                  <input className="text" id="topic1" type="text" placeholder="group topic 1" />
                </label>
              </div>
              <div className="textDiv" id="topic2Div">
                <label htmlFor="groupdescription">
                  Topic 2:
                  <input className="text" id="topic2" type="text" placeholder="group topic 2" />
                </label>
              </div>
              <div className="textDiv" id="topic-3-Div">
                <label htmlFor="ispublic">
                  Topic 3:
                  <input className="text" id="topic3" type="text" placeholder="group topic 3" />
                </label>
              </div>
            </div>

            <button type="submit" className="createGroup" onClick={() => { createGroup(); updateGroups(); }}> Create A Group </button>
            <div className="text-area" />
            <div className="post-button" />
          </div>

          <div className="groups-area" id="groups-area">
            Groups will be displayed here
            <div className="box-1">
              <div className="poster-picture" />
              <div className="update-text" />
              <div className="reply-button" />
            </div>

            <div className="box-2">
              <div className="poster-picture" />
              <div className="update-text" />
              <div className="reply-button" />
            </div>
          </div>
        </div>

        <div className="message-updates">
          Message updates
        </div>

      </div>
    </div>
  );
}

export default Groups;
