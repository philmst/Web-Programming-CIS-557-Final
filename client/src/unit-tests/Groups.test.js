import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Groups from '../groups-page/Groups';

/**
 * @jest-environment jsdom
 */

const changeLink = jest.fn();

describe('Test Group Page UI', () => {
  test('Group page renders correctly', () => {
    const component = renderer.create(<Groups changeState={changeLink} state={{username: 'me'}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});