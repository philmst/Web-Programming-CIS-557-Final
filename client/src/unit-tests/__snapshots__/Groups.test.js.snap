// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Group Page UI Group page renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, me!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        onClick={[Function]}
        type="submit"
      >
        Analytics
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="invitations"
        onClick={[Function]}
        type="submit"
      >
        Group Invitations and Requests 
      </button>
      <button
        className="photos"
        onClick={[Function]}
        type="submit"
      >
        Flagged posts
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="post-area"
      >
        <div
          className="left-group-input"
        >
          <div
            className="textDiv"
            id="groupnameDiv"
          >
            <label
              htmlFor="groupname"
            >
              Group Name:
              <input
                className="text"
                id="groupname"
                placeholder="group name"
                type="text"
              />
            </label>
          </div>
          <div
            className="textDiv"
            id="groupdescriptionDiv"
          >
            <label
              htmlFor="groupdescription"
            >
              Group Description:
              <input
                className="text"
                id="groupdescription"
                placeholder="group description"
                type="text"
              />
            </label>
          </div>
          <div
            className="textDiv"
            id="ispublicDiv"
          >
            <label
              htmlFor="ispublic"
            >
              Is Public?
              <input
                className="text"
                id="ispublic"
                placeholder="1 or 0"
                type="text"
              />
            </label>
          </div>
        </div>
        <div
          className="right-group-input"
        >
          <div
            className="textDiv"
            id="topic1Div"
          >
            <label
              htmlFor="topic1"
            >
              Topic 1:
              <input
                className="text"
                id="topic1"
                placeholder="group topic 1"
                type="text"
              />
            </label>
          </div>
          <div
            className="textDiv"
            id="topic2Div"
          >
            <label
              htmlFor="groupdescription"
            >
              Topic 2:
              <input
                className="text"
                id="topic2"
                placeholder="group topic 2"
                type="text"
              />
            </label>
          </div>
          <div
            className="textDiv"
            id="topic-3-Div"
          >
            <label
              htmlFor="ispublic"
            >
              Topic 3:
              <input
                className="text"
                id="topic3"
                placeholder="group topic 3"
                type="text"
              />
            </label>
          </div>
        </div>
        <button
          className="createGroup"
          onClick={[Function]}
          type="submit"
        >
           Create A Group 
        </button>
      </div>
      <div
        className="heading-for-groups"
        id="heading-for-groups"
      >
        Groups in which you're a member:
      </div>
      <div
        className="groups-area"
        id="groups-area"
      >
        There are no groups yet!
      </div>
      <div
        className="groups-non-member-message"
        id="groups-non-member-message"
      >
         
      </div>
      <div
        className="heading-for-groups"
        id="heading-for-groups-non-member"
      >
        Other public groups you can request to join:
      </div>
      <div
        className="groups-area"
        id="groups-area-non-member"
      >
        <p />
      </div>
      <div
        className="heading-for-groups"
        id="heading-for-suggested-group"
      >
        Here's a heavily personalised (not randomised at all) group recommendation:
      </div>
      <div
        className="groups-area"
        id="groups-area-suggestions"
      />
    </div>
    <div
      className="side-navbar"
      id="forMessages"
    >
      <button
        className="messages"
        onClick={[Function]}
        type="submit"
      >
        Messages
      </button>
      <button
        className="logout"
        onClick={[Function]}
        type="submit"
      >
        Log out
      </button>
    </div>
  </div>
</div>
`;
